secboot/secboot_tpm_test.go

```diff
@@ -59,75 +59,87 @@ func (s *secbootSuite) TestCheckKeySealingSupported(c *C) {
 	sbEnabled := []uint8{1}
 	sbDisabled := []uint8{0}
 	efiNotSupported := []uint8(nil)
+	tpmErr := errors.New("TPM error")
 
 	type testCase struct {
-		hasTPM bool
-		sbData []uint8
-		errStr string
+		tpmErr     error
+		tpmEnabled bool
+		sbData     []uint8
+		err        string
 	}
-	for _, t := range []testCase{
-		{true, sbEnabled, ""},
-		{true, sbEmpty, "secure boot variable does not exist"},
-		{false, sbEmpty, "secure boot variable does not exist"},
-		{true, sbDisabled, "secure boot is disabled"},
-		{false, sbEnabled, "cannot connect to TPM device: TPM not available"},
-		{false, sbDisabled, "secure boot is disabled"},
-		{true, efiNotSupported, "not a supported EFI system"},
+	for i, tc := range []testCase{
+		// happy case
+		{tpmErr: nil, tpmEnabled: true, sbData: sbEnabled, err: ""},
+		// secure boot EFI var is empty
+		{tpmErr: nil, tpmEnabled: true, sbData: sbEmpty, err: "secure boot variable does not exist"},
+		// secure boot is disabled
+		{tpmErr: nil, tpmEnabled: true, sbData: sbDisabled, err: "secure boot is disabled"},
+		// EFI not supported
+		{tpmErr: nil, tpmEnabled: true, sbData: efiNotSupported, err: "not a supported EFI system"},
+		// TPM connection error
+		{tpmErr: tpmErr, sbData: sbEnabled, err: "cannot connect to TPM device: TPM error"},
+		// TPM was detected but it's not enabled
+		{tpmErr: nil, tpmEnabled: false, sbData: sbEnabled, err: "TPM device is not enabled"},
```

Comment:
Should we have a test case with `sb.ErrNoTPM2Device` to see if it's surfaced up?